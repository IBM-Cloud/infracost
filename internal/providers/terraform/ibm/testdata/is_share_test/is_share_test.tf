terraform {
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = "1.63.0"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_is_vpc" "testVpc" {
  name = "test-vpc"
}

resource "ibm_is_share" "testNFS" {
  name    = "test-share"
  size    = 100
  profile = "dp2"
  zone    = "us-south-2"
  iops    = 100
}

resource "ibm_is_share" "testReplica" {
  zone                  = "us-south-3"
  source_share          = ibm_is_share.testNFS.id
  name                  = "test-replica-share"
  profile               = "dp2"
  replication_cron_spec = "0 */5 * * *"
}

resource "ibm_is_share_mount_target" "testNFSMount" {
  share = ibm_is_share.testNFS.id
  vpc   = ibm_is_vpc.testVpc.id
  name  = "test-share-target"
}

resource "ibm_is_share" "testNFS2withInlineReplica" {
  name    = "test-share-2"
  size    = 200
  profile = "dp2"
  zone    = "us-south-2"

  replica_share {
    name                  = "test-replica-share-2"
    replication_cron_spec = "0 */5 * * *"
    profile               = "dp2"
    zone                  = "us-south-3"
  }
}
