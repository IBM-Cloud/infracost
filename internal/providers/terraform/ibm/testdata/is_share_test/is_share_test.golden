
 Name                                                              Monthly Qty  Unit                      Monthly Cost 
                                                                                                                       
 ibm_is_share.testNFS                                                                                                  
 ├─ 100 GB file storage share for VPC - us-south-2                      73,000  GB Hours                        $15.00 
 └─ 100 Max IOPS (first 5000 IOPS Hours)                                 5,000  IOPS Hours                       $0.69 
 └─ 100 Max IOPS (next 15000 IOPS Hours)                                15,000  IOPS Hours                       $1.03 
 └─ 100 Max IOPS (next 20000 IOPS Hours)                                20,000  IOPS Hours                       $0.55 
 └─ 100 Max IOPS (over 40000 IOPS Hours)                                33,000  IOPS Hours                       $0.45 
                                                                                                                       
 ibm_is_share.testNFS2withInlineReplica                                                                                
 ├─ 200 GB file storage share for VPC - us-south-2                     146,000  GB Hours                        $30.00 
 ├─ 0 Max IOPS (first 5000 IOPS Hours)                      Monthly cost depends on usage: $0.000137 per IOPS Hours    
 ├─ 0 Max IOPS (next 15000 IOPS Hours)                      Monthly cost depends on usage: $0.0000685 per IOPS Hours   
 ├─ 0 Max IOPS (next 20000 IOPS Hours)                      Monthly cost depends on usage: $0.0000274 per IOPS Hours   
 ├─ 0 Max IOPS (over 40000 IOPS Hours)                      Monthly cost depends on usage: $0.0000137 per IOPS Hours   
 ├─ 200 GB Replica file storage share for VPC - us-south-3             146,000  GB Hours                        $30.00 
 ├─ 0 Max IOPS (first 5000 IOPS Hours)                      Monthly cost depends on usage: $0.000137 per IOPS Hours    
 ├─ 0 Max IOPS (next 15000 IOPS Hours)                      Monthly cost depends on usage: $0.0000685 per IOPS Hours   
 ├─ 0 Max IOPS (next 20000 IOPS Hours)                      Monthly cost depends on usage: $0.0000274 per IOPS Hours   
 ├─ 0 Max IOPS (over 40000 IOPS Hours)                      Monthly cost depends on usage: $0.0000137 per IOPS Hours   
 └─ Transmitted data                                                         1  GB                               $0.02 
                                                                                                                       
 ibm_is_share.testReplica                                                                                              
 ├─ 100 GB Replica file storage share for VPC - us-south-3  Monthly cost depends on usage: $0.0002055 per GB Hours     
 ├─ 0 Max IOPS (first 5000 IOPS Hours)                      Monthly cost depends on usage: $0.000137 per IOPS Hours    
 ├─ 0 Max IOPS (next 15000 IOPS Hours)                      Monthly cost depends on usage: $0.0000685 per IOPS Hours   
 ├─ 0 Max IOPS (next 20000 IOPS Hours)                      Monthly cost depends on usage: $0.0000274 per IOPS Hours   
 ├─ 0 Max IOPS (over 40000 IOPS Hours)                      Monthly cost depends on usage: $0.0000137 per IOPS Hours   
 └─ Transmitted data                                        Monthly cost depends on usage: $0.02 per GB                
                                                                                                                       
 ibm_is_vpc.testVpc                                                                                                    
 ├─ VPC instance                                                             1  Instance                         $0.00 
 ├─ VPC egress free allowance (first 5GB)                   Monthly cost depends on usage: $0.00 per GB                
 └─ VPC egress us-south (first 9995 GB)                     Monthly cost depends on usage: $0.090915 per GB            
 └─ VPC egress us-south (next 40000 GB)                     Monthly cost depends on usage: $0.086735 per GB            
 └─ VPC egress us-south (next 100000 GB)                    Monthly cost depends on usage: $0.07315 per GB             
 └─ VPC egress us-south (over 149995 GB)                    Monthly cost depends on usage: $0.05225 per GB             
                                                                                                                       
 OVERALL TOTAL                                                                                                  $77.74 
──────────────────────────────────
5 cloud resources were detected:
∙ 4 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 1 was free:
  ∙ 1 x ibm_is_share_mount_target